apps/site/lib/site_web/controllers/schedule/finder_api.ex

```diff
@@ -0,0 +1,370 @@
+defmodule SiteWeb.ScheduleController.FinderApi do
+  @moduledoc """
+    API for retrieving journeys for a route, and for
+    showing trip details for each journey on demand.
+
+    "Departure" here is analagous to PredictedSchedule.t()
+  """
+  use SiteWeb, :controller
+
+  alias Predictions.Prediction
+  alias Routes.Route
+  alias Schedules.{Schedule, Trip}
+  alias Site.TransitNearMe
+  alias SiteWeb.ScheduleController.TripInfo, as: Trips
+  alias SiteWeb.ScheduleController.VehicleLocations, as: Vehicles
+
+  import SiteWeb.ScheduleController.ScheduleApi, only: [format_time: 1, fares_for_service: 4]
+
+  @type react_keys :: :date | :direction | :is_current
+  @type react_strings :: [{react_keys, String.t()}]
+  @type converted_values :: {Date.t(), integer, boolean}
+
+  @type enhanced_journey :: %{
+          departure: PredictedSchedule.t() | nil,
+          arrival: PredictedSchedule.t() | nil,
+          trip: Trip.t() | nil,
+          realtime: TransitNearMe.time_data() | nil
+        }
+
+  # How many seconds a departure is considered recent
+  @recent_departure_retention_policy 600
+
+  # Leverage the JourneyList module to return a simplified set of trips
+  @spec journeys(Plug.Conn.t(), map) :: Plug.Conn.t()
+  def journeys(conn, %{"stop" => stop_id, "date" => date} = params) do
+    {:ok, user_selected_date} = Date.from_iso8601(date)
+    {schedules, predictions} = load_from_repos(conn, params)
+
+    # Emulate original Schedules tab journey build logic:
```

Comment:
Explaining here _why_ we're doing this might be useful to future developers.

Reply:
The original Schedules tab contained a complete list of trips (JourneyList) (minus detailed stop info) for the date, and then allowed to you drill down into a specific trip (TripInfo) on demand. These structs contained all the information Ingrid desired in her designs, matching the design spec architecturally better than the existing Schedules-only and Predictions-only configuration for ScheduleFinder.